### ISSUE TITLE
[CURATOR-270] createContainers does not work correctly with usingNamespace

### ISSUE THREAD

**Issue by jira-importer** [apache/infrastructure-jira-issues-import-test-curator#269]
<p>I just recently upgraded my project to 2.9.0 and noticed that <tt>PathChildrenCache</tt> recipe started failing with</p>

<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>Caused by: org.apache.zookeeper.KeeperException$NoNodeException: KeeperErrorCode = NoNode for /parent
at org.apache.zookeeper.KeeperException.create(KeeperException.java:111) ~[zookeeper-3.4.6.jar:3.4.6-1569965]
at org.apache.zookeeper.KeeperException.create(KeeperException.java:51) ~[zookeeper-3.4.6.jar:3.4.6-1569965]
at org.apache.zookeeper.ZooKeeper.getChildren(ZooKeeper.java:1590) ~[zookeeper-3.4.6.jar:3.4.6-1569965]
at org.apache.curator.framework.imps.GetChildrenBuilderImpl$3.call(GetChildrenBuilderImpl.java:214) ~[curator-framework-2.9.0.jar:?]
at org.apache.curator.framework.imps.GetChildrenBuilderImpl$3.call(GetChildrenBuilderImpl.java:203) ~[curator-framework-2.9.0.jar:?]
at org.apache.curator.RetryLoop.callWithRetry(RetryLoop.java:107) ~[curator-client-2.9.0.jar:?]
at org.apache.curator.framework.imps.GetChildrenBuilderImpl.pathInForeground(GetChildrenBuilderImpl.java:200) ~[curator-framework-2.9.0.jar:?]
at org.apache.curator.framework.imps.GetChildrenBuilderImpl.forPath(GetChildrenBuilderImpl.java:191) ~[curator-framework-2.9.0.jar:?]
at org.apache.curator.framework.imps.GetChildrenBuilderImpl.forPath(GetChildrenBuilderImpl.java:38) ~[curator-framework-2.9.0.jar:?]
at org.apache.curator.framework.recipes.cache.PathChildrenCache.rebuild(PathChildrenCache.java:326) ~[curator-recipes-2.9.0.jar:?]
at org.apache.curator.framework.recipes.cache.PathChildrenCache.start(PathChildrenCache.java:299) ~[curator-recipes-2.9.0.jar:?]
</pre>
</div></div>

<p>It turns out that <tt>PathChildrenCache</tt> recently started using new implementation <tt>createContainers</tt> for creating parent directories/znodes and this method has a bug in <tt>NamespaceFacade.createContainers</tt> implementation, because it does not account for namespace name.</p>

<p>I'll create a PR with a test and fix shortly.</p>

---
<details><summary><i>Originally reported by <a title="Alexey Serba" href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=alexey.serba">alexey.serba</a>, imported from: <a href="https://issues.apache.org/jira/browse/CURATOR-270" target="_blank">createContainers does not work correctly with usingNamespace</a></i></summary>
<i><ul>
<li><b>assignee</b>: <a title="Jordan Zimmerman" href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=randgalt">randgalt</a>
<li><b>status</b>: Resolved
<li><b>priority</b>: Major
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2015-10-06T14:23:03+01:00
<li><b>imported</b>: 2025-01-19
</ul></i>
</details>

**Comment by jira-importer** [comment]
<i><a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=githubbot">githubbot</a>:</i>
<p>GitHub user serba opened a pull request:</p>

<p>    <a href="https://github.com/apache/curator/pull/113" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/apache/curator/pull/113</a></p>

<p>    <a href="https://issues.apache.org/jira/browse/CURATOR-270" title="createContainers does not work correctly with usingNamespace" class="issue-link" data-issue-key="CURATOR-270"><del>CURATOR-270</del></a> createContainers does not work correctly with usingNamespace</p>

<p>    I just recently upgraded my project to 2.9.0 and noticed that `PathChildrenCache` recipe started failing with</p>

<p>    ```<br/>
    Caused by: org.apache.zookeeper.KeeperException$NoNodeException: KeeperErrorCode = NoNode for /parent<br/>
    at org.apache.zookeeper.KeeperException.create(KeeperException.java:111) ~<span class="error">&#91;zookeeper-3.4.6.jar:3.4.6-1569965&#93;</span><br/>
    at org.apache.zookeeper.KeeperException.create(KeeperException.java:51) ~<span class="error">&#91;zookeeper-3.4.6.jar:3.4.6-1569965&#93;</span><br/>
    at org.apache.zookeeper.ZooKeeper.getChildren(ZooKeeper.java:1590) ~<span class="error">&#91;zookeeper-3.4.6.jar:3.4.6-1569965&#93;</span><br/>
    at org.apache.curator.framework.imps.GetChildrenBuilderImpl$3.call(GetChildrenBuilderImpl.java:214) ~<span class="error">&#91;curator-framework-2.9.0.jar:?&#93;</span><br/>
    at org.apache.curator.framework.imps.GetChildrenBuilderImpl$3.call(GetChildrenBuilderImpl.java:203) ~<span class="error">&#91;curator-framework-2.9.0.jar:?&#93;</span><br/>
    at org.apache.curator.RetryLoop.callWithRetry(RetryLoop.java:107) ~<span class="error">&#91;curator-client-2.9.0.jar:?&#93;</span><br/>
    at org.apache.curator.framework.imps.GetChildrenBuilderImpl.pathInForeground(GetChildrenBuilderImpl.java:200) ~<span class="error">&#91;curator-framework-2.9.0.jar:?&#93;</span><br/>
    at org.apache.curator.framework.imps.GetChildrenBuilderImpl.forPath(GetChildrenBuilderImpl.java:191) ~<span class="error">&#91;curator-framework-2.9.0.jar:?&#93;</span><br/>
    at org.apache.curator.framework.imps.GetChildrenBuilderImpl.forPath(GetChildrenBuilderImpl.java:38) ~<span class="error">&#91;curator-framework-2.9.0.jar:?&#93;</span><br/>
    at org.apache.curator.framework.recipes.cache.PathChildrenCache.rebuild(PathChildrenCache.java:326) ~<span class="error">&#91;curator-recipes-2.9.0.jar:?&#93;</span><br/>
    at org.apache.curator.framework.recipes.cache.PathChildrenCache.start(PathChildrenCache.java:299) ~<span class="error">&#91;curator-recipes-2.9.0.jar:?&#93;</span><br/>
    ```</p>

<p>    It turns out that `PathChildrenCache` recently started using new implementation `createContainers` for creating parent directories/znodes and this method has a bug in `NamespaceFacade.createContainers` implementation, because it does not account for namespace name.</p>


<p>You can merge this pull request into a Git repository by running:</p>

<p>    $ git pull <a href="https://github.com/serba/curator" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/serba/curator</a> <a href="https://issues.apache.org/jira/browse/CURATOR-270" title="createContainers does not work correctly with usingNamespace" class="issue-link" data-issue-key="CURATOR-270"><del>CURATOR-270</del></a></p>

<p>Alternatively you can review and apply these changes as the patch at:</p>

<p>    <a href="https://github.com/apache/curator/pull/113.patch" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/apache/curator/pull/113.patch</a></p>

<p>To close this pull request, make a commit to your master/trunk branch<br/>
with (at least) the following in the commit message:</p>

<p>    This closes #113</p>

<hr />
<p>commit 1556a2fcb8fd0d43669b057eb51290e3bf5ee5a2<br/>
Author: Alexey Serba <aserba@gmail.com><br/>
Date:   2015-10-05T21:11:35Z</p>

<p>    <a href="https://issues.apache.org/jira/browse/CURATOR-270" title="createContainers does not work correctly with usingNamespace" class="issue-link" data-issue-key="CURATOR-270"><del>CURATOR-270</del></a> createContainers does not work correctly with usingNamespace</p>

<hr />

**Comment by jira-importer** [comment]
<i><a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=githubbot">githubbot</a>:</i>
<p>Github user asfgit closed the pull request at:</p>

<p>    <a href="https://github.com/apache/curator/pull/113" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/apache/curator/pull/113</a></p>
